cms/djangoapps/modulestore_migrator/tasks.py

```diff
@@ -285,11 +299,12 @@ def migrate_from_modulestore(
     # a given LearningPackage.
     # We use this mapping to ensure that we don't create duplicate
     # PublishableEntities during the migration process for a given LearningPackage.
-    existing_source_to_target_keys = {
-        block.source.key: block.target for block in ModulestoreBlockMigration.objects.filter(
-            overall_migration__target=migration.target.id
-        )
```

Comment:
The underlying bug was here. It was only considering the last migration, so after migrating the second time, the slug used for the first migration would appear available and reused (causing an update).